napytau/gui/gui_app.py

```diff
@@ -0,0 +1,478 @@
+from typing import List, Tuple
+
+import tkinter as tk
+from tkinter import Canvas
+from tkinter import filedialog
+
+import customtkinter
+from matplotlib.figure import Figure
+from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg
+
+from napytau.cli.cli_arguments import CLIArguments
+from napytau.core.logic_mockup import logic
+from tkinter import Menu
+
+from napytau.gui.checkbox_datapoint import CheckboxDataPoint
+
+customtkinter.set_appearance_mode(
+    "System"
+)  # Modes: "System" (standard), "Dark", "Light"
+customtkinter.set_default_color_theme(
+    "blue"
+)  # Themes: "blue" (standard), "green", "dark-blue"
+
+
+# Modes: "System" (standard), "Dark", "Light"
+customtkinter.set_appearance_mode("System")
+# Themes: "blue" (standard), "green", "dark-blue"
+customtkinter.set_default_color_theme("blue")
+
+class GUIApp(customtkinter.CTk):
+    def __init__(self) -> None:
+        """
+        Constructor for the GUIApp, initializes the GUI.
```

Comment:
I did this on purpose because I didn't want to over-modularize the code.
If for example the menubar is a separate class, one could argue to also refactor the buttons within the menubar (like File, View, ...) into separate classes and also the options within these buttons.
Then you end up with a lot of classes which are all initialized and used once in the same spot.

But I also see your point, especially for the graph.
What is your opinion on splitting up the GUI only into the classes MenuBar, Graph, Informationfield and Checkboxfield (names more or less like that)?
I would refactor the menubar and checkboxes which this PR/Issue is about and leave the graph and info-field for a separate Issue/PR.